src/librustc_save_analysis/lib.rs

```diff
@@ -533,14 +533,17 @@ impl<'l, 'tcx> SaveContext<'l, 'tcx> {
                 match self.tables.expr_ty_adjusted(&hir_node).kind {
                     ty::Adt(def, _) if !def.is_enum() => {
                         let variant = &def.non_enum_variant();
-                        let index = self.tcx.find_field_index(ident, variant).unwrap();
-                        filter!(self.span_utils, ident.span);
-                        let span = self.span_from_span(ident.span);
-                        return Some(Data::RefData(Ref {
-                            kind: RefKind::Variable,
-                            span,
-                            ref_id: id_from_def_id(variant.fields[index].did),
-                        }));
+                        if let Some(index) = self.tcx.find_field_index(ident, variant) {
```

Comment:
Could we use `map` here? Since we indent by a level anyway and to avoid trailing `None`

Reply:
Addressed via a211a82.